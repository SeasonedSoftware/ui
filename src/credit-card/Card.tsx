import React from 'react'
import PropTypes from 'prop-types'

import Cards from 'react-credit-cards'
import 'react-credit-cards/es/styles-compiled.css'

const Card = ({ formState, focused, acceptedCards }:any) => {
  const cardProps = { ...formState.values, focused, acceptedCards }
  return <Cards {...cardProps} />
}

Card.propTypes = {
  focused: PropTypes.string,
  acceptedCards: PropTypes.arrayOf(PropTypes.string)
}

export default Card
