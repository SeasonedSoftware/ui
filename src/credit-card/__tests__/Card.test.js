import React from 'react'
import renderer from 'react-test-renderer'

import Card from '../Card'

jest.mock('react-credit-cards', () => props => <div {...props}>Cards</div>)

it('renders correctly', () => {
  const props = {
    formState: {
      values: {
        number: '',
      },
    },
    focused: 'name',
  }

  const tree = renderer.create(<Card {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
