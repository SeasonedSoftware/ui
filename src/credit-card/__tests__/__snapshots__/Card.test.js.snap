// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  focused="name"
  number=""
>
  Cards
</div>
`;
